x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/domains_table.test.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockActions, setMockValues } from '../../../../__mocks__/kea_logic';
+import '../../../__mocks__/engine_logic.mock';
+
+import React from 'react';
+
+import { shallow, ShallowWrapper } from 'enzyme';
+
+import { EuiBasicTable, EuiButtonIcon, EuiInMemoryTable } from '@elastic/eui';
+
+import { KibanaLogic } from '../../../../shared/kibana';
+import { mountWithIntl } from '../../../../test_helpers';
+
+import { CrawlerDomain } from '../types';
+
+import { DomainsTable } from './domains_table';
+
+const domains: CrawlerDomain[] = [
+  {
+    id: '1234',
+    documentCount: 9999,
+    url: 'elastic.co',
+    crawlRules: [],
+    entryPoints: [],
+    sitemaps: [],
+    lastCrawl: '2020-01-01T00:00:00-05:00',
+    createdOn: '2020-01-01T00:00:00-05:00',
+  },
+];
+
+const values = {
+  // EngineLogic
+  engineName: 'some-engine',
+  // CrawlerOverviewLogic
+  domains,
+  // AppLogic
+  myRole: { canManageEngineCrawler: false },
+};
+
+const actions = {
+  // CrawlerOverviewLogic
+  deleteDomain: jest.fn(),
+};
+
+describe('DomainsTable', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(values);
+    setMockActions(actions);
+  });
+
+  it('renders', () => {
+    const wrapper = shallow(<DomainsTable />);
+    expect(wrapper.find(EuiInMemoryTable)).toHaveLength(1);
+  });
+
+  describe('columns', () => {
+    let wrapper: ShallowWrapper;
+    let tableContent: string;
+
+    const getTable = () => wrapper.find(EuiInMemoryTable).dive().find(EuiBasicTable).dive();
+
+    beforeEach(() => {
+      wrapper = shallow(<DomainsTable />);
+      tableContent = mountWithIntl(<DomainsTable />)
+        .find(EuiInMemoryTable)
+        .text();
+    });
```

Comment:
[test perf] Since we're never actually changing any Kea values or logic in this test file, IMO all of this can be set in a single `beforeAll` instead of `beforeEach` so that we're only mounting once instead of 5+ times.

```suggestion
  let wrapper: ShallowWrapper;
  let tableContent: string;

  beforeAll(() => {
    setMockValues(values);
    setMockActions(actions);
    wrapper = shallow(<DomainsTable />);
    tableContent = mountWithIntl(<DomainsTable />)
      .find(EuiInMemoryTable)
      .text();
  });

  it('renders', () => {
    expect(wrapper.find(EuiInMemoryTable)).toHaveLength(1);
  });

  describe('columns', () => {
    const getTable = () => wrapper.find(EuiInMemoryTable).dive().find(EuiBasicTable).dive();
```

Reply:
I've made this change in d1ae5e9